openhands/utils/prompt.py

```diff
@@ -63,11 +65,23 @@ def get_example_user_message(self, latest_user_message: str) -> str:
         These additional context will convert the current generic agent
         into a more specialized agent that is tailored to the user's task.
         """
+        return self.user_template.render().strip()
+
+    def enhance_message(self, message: Message, state: State) -> None:
+        """Enhance the user message with additional context.
+
+        This method is used to enhance the user message with additional context
+        about the user's task. The additional context will convert the current
+        generic agent into a more specialized agent that is tailored to the user's task.
+        """
         micro_agent_prompts = []
         for micro_agent in self.microagents.values():
-            if micro_agent.should_trigger(latest_user_message):
+            if micro_agent.should_trigger(message):
                 micro_agent_prompts.append(micro_agent.content)
-        rendered = self.user_template.render(
-            micro_agents=micro_agent_prompts,
-        )
-        return rendered.strip()
+        if len(micro_agent_prompts) > 0:
+            micro_text = "EXTRA INFO: the following information has been included based on a keyword match. It may or may not be relevant to the user's request.\n\n"
```

Comment:
This may sound like magical thinking, but since we're at it on other branches: the Anthropic prompts everywhere seem to show heavy use of xml tags. While this isn't exactly news, the latest bunch of reveals seems worth trying more and hey they probably don't hurt anyway. We could add this like `<EXTRA_INFO> </EXTRA_INFO>` ?

On a side note, I feel like it's becoming more important to figure out what to do with other LLMs... when optimizing for one, it perhaps shouldn't come as a surprise that it seems to win "big". I mean, I'm not talking about tool use; that one, I think, may yet prove very good for all LLMs that support it, but apart from tool use itself, we also do stuff like this. ^ 🤷

Reply:
good idea! fixed